src/java.base/share/classes/java/io/Console.java

```diff
@@ -45,8 +45,9 @@
  * output streams then its console will exist and will typically be
  * connected to the keyboard and display from which the virtual machine
  * was launched.  If the virtual machine is started automatically, for
- * example by a background job scheduler, then it may not
- * have a console.
+ * example by a background job scheduler, or if one or both of the
+ * standard input or output has been redirected, then it may not have a
```

Comment:
If we are claiming that `System.console()` should only return a "Console instance only when both standard input and output are connected to a terminal." then is this _may_ wording not restrictive enough? Or is that nuance negligible?

Reply:
Intention here is that, we do want to restirct to not return JLine based Console when either/both of stdin/out is redirected, I think we would not want to restrict them as the spec. For example, `System.console()` in jshell returns non-null, even if the remote JVM is not-attached to a terminal. I will describe this sublety in the CSR